internal/cli/envs/env_checker.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envs
+
+import "os"
+
+type EnvChecker interface {
+	IsPopulated(env string) bool
+}
+
+type NewEnvChecker struct{}
```

Comment:
Yep because when running the tests without mocking the env var `ATLAS_GITHUB_ACTION` is set which causes tests to fail when run in the GitHub CI

I couldn't see another way around it and imo it makes sense not to set/check env vars directly in tests